src/screens/Settings/HSwitchProfile/TwoFaSettings/RegenerateRC.res

```diff
@@ -0,0 +1,185 @@
+let p3Regular = HSwitchUtils.getTextClass((P3, Regular))
+
+@react.component
+let make = (~checkStatusResponse: HSwitchSettingTypes.checkStatusType) => {
+  open LogicUtils
+  open APIUtils
+
+  let getURL = useGetURL()
+  let fetchDetails = useGetMethod()
+  let showToast = ToastState.useShowToast()
+  let verifyTotpLogic = TotpHooks.useVerifyTotp()
+  let (showVerifyModal, setShowVerifyModal) = React.useState(_ => false)
+  let (otpInModal, setOtpInModal) = React.useState(_ => "")
+  let (buttonState, setButtonState) = React.useState(_ => Button.Normal)
+  let (recoveryCodes, setRecoveryCodes) = React.useState(_ => [])
+  let (errorMessage, setErrorMessage) = React.useState(_ => "")
+  let (screenState, setScreenState) = React.useState(_ => PageLoaderWrapper.Success)
+
+  let generateRecoveryCodes = async () => {
+    try {
+      setScreenState(_ => PageLoaderWrapper.Loading)
+      let url = getURL(~entityName=USERS, ~userType=#GENERATE_RECOVERY_CODES, ~methodType=Get)
+      let response = await fetchDetails(url)
+      let recoveryCodesValue = response->getDictFromJsonObject->getStrArray("recovery_codes")
+      setRecoveryCodes(_ => recoveryCodesValue)
+      setScreenState(_ => PageLoaderWrapper.Success)
+    } catch {
+    | _ => {
+        setButtonState(_ => Button.Normal)
+        showToast(~message="Failed to generate recovery codes!", ~toastType=ToastError)
+        RescriptReactRouter.push(GlobalVars.appendDashboardPath(~url=`/account-settings/profile`))
+        setScreenState(_ => PageLoaderWrapper.Success)
+      }
+    }
+  }
+
+  let handleModalClose = () => {
+    RescriptReactRouter.push(GlobalVars.appendDashboardPath(~url=`/account-settings/profile`))
+  }
+
+  let verifyTOTP = async () => {
+    try {
+      setButtonState(_ => Button.Loading)
+      if otpInModal->String.length > 0 {
+        let body = [("totp", otpInModal->JSON.Encode.string)]->getJsonFromArrayOfJson
+        let _ = await verifyTotpLogic(body, Post)
+        setShowVerifyModal(_ => false)
+        generateRecoveryCodes()->ignore
+      } else {
+        showToast(~message="OTP field cannot be empty!", ~toastType=ToastError)
+      }
+      setOtpInModal(_ => "")
+      setButtonState(_ => Button.Normal)
+    } catch {
+    | Exn.Error(e) => {
+        let err = Exn.message(e)->Option.getOr("Verification Failed")
+        let errorMessage = err->safeParse->getDictFromJsonObject->getString("message", "")
+        setOtpInModal(_ => "")
+        setErrorMessage(_ => errorMessage)
+        setButtonState(_ => Button.Normal)
+      }
+    }
+  }
+
+  React.useEffect(() => {
+    if checkStatusResponse.totp {
+      generateRecoveryCodes()->ignore
+    } else {
+      setShowVerifyModal(_ => true)
+    }
+    None
+  }, [])
+
+  let handleKeyUp = ev => {
+    open ReactEvent.Keyboard
+    let key = ev->key
+    let keyCode = ev->keyCode
+
+    if key === "Enter" || keyCode === 13 {
+      verifyTOTP()->ignore
+    }
+  }
```

Comment:
Same just check once

Reply:
By using both checks (key === 'Enter' || keyCode === 13), we ensure that your application works across all browsers, and for older and newer versions